client/src/Components/privateRoute.jsx

```diff
@@ -0,0 +1,21 @@
+import React, {useContext} from 'react';
+import { Route, Redirect } from 'react-router-dom';
+import { UserContext } from '../context/userContext.js'
+
+
+const PrivateRoute = ({component: Component, ...rest}) => {
+    const {user} = useContext(UserContext)
+
+    return (
+
+        // Show the component only when the user is logged in
+        // Otherwise, redirect the user to /signin page
+        <Route {...rest} render={props => (
+            user.username ?
+                <Component {...props} />
+            : <Redirect to="/login" />
```

Comment:
The full beauty of conditional route rendering / redirecting in action. Nicely secured your holy routes.